distribution/packages/build.gradle

```diff
@@ -203,37 +146,166 @@ subprojects {
-    into(new File(project.expansions['path.env']).getParent()) {
-      fileType CONFIG | NOREPLACE
-      fileMode 0660
-      from "${project.packagingFiles}/env/elasticsearch"
-    }
 
-    /**
-     * Suck up all the empty directories that we need to install into the path.
-     */
-    Closure suckUpEmptyDirectories = { path, u, g, mode ->
-      into(path) {
+    // ========= empty dirs =========
+    // NOTE: these are created under packagingFiles as empty, but the permissions are set here
+    Closure copyEmptyDir = { path, u, g, mode ->
+      File file = new File(path)
+      into(file.parent) {
         from "${packagingFiles}/${path}"
+        include file.name
         includeEmptyDirs true
         createDirectoryEntry true
         user u
         permissionGroup g
         dirMode mode
-        fileMode mode
       }
     }
-    suckUpEmptyDirectories('/var/run', 'elasticsearch', 'elasticsearch', 0755)
-    suckUpEmptyDirectories('/var/log', 'elasticsearch', 'elasticsearch', 0750)
-    suckUpEmptyDirectories('/var/lib', 'elasticsearch', 'elasticsearch', 0750)
-    suckUpEmptyDirectories('/usr/share/elasticsearch', 'root', 'root', 0755)
+    copyEmptyDir('/var/run/elasticsearch', 'elasticsearch', 'elasticsearch', 0755)
+    copyEmptyDir('/var/log/elasticsearch', 'elasticsearch', 'elasticsearch', 0750)
+    copyEmptyDir('/var/lib/elasticsearch', 'elasticsearch', 'elasticsearch', 0750)
+    copyEmptyDir('/usr/share/elasticsearch/plugins', 'root', 'root', 0755)
+  }
+}
+
+apply plugin: 'nebula.ospackage-base'
+
+// this is package indepdendent configuration
+ospackage {
+  packageName 'elasticsearch'
+  maintainer 'Elasticsearch Team <info@elastic.co>'
+  summary '''
+    Elasticsearch is a distributed RESTful search engine built for the cloud.
+    Reference documentation can be found at
+    https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html
+    and the 'Elasticsearch: The Definitive Guide' book can be found at
+    https://www.elastic.co/guide/en/elasticsearch/guide/current/index.html
+    '''.stripIndent().replace('\n', ' ').trim()
+  url 'https://www.elastic.co/'
+
+  // signing setup
+  if (project.hasProperty('signing.password') && System.getProperty('build.snapshot', 'true') == 'false') {
+    signingKeyId = project.hasProperty('signing.keyId') ? project.property('signing.keyId') : 'D88E42B4'
+    signingKeyPassphrase = project.property('signing.password')
+    signingKeyRingFile = project.hasProperty('signing.secretKeyRingFile') ?
+                         project.file(project.property('signing.secretKeyRingFile')) :
+                         new File(new File(System.getProperty('user.home'), '.gnupg'), 'secring.gpg')
+  }
+
+  requires('coreutils')
+
+  fileMode 0644
+  dirMode 0755
+  user 'root'
+  permissionGroup 'root'
+
+  into '/usr/share/elasticsearch'
+  with libFiles
+  with modulesFiles
+  with noticeFile
+}
+
+task buildDeb(type: Deb) {
```

Comment:
I think it is worth commenting that these are configured on this project but are part of the packaging for the subprojects.